From ed428ddfef87a5641598a017e8781f107f3f3c27 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Thu, 19 Aug 2021 16:15:13 +0100
Subject: [PATCH] arm: nStackBytes was already the right amount of bytes

possible regression from...

commit 6e3424ca1131fe371f63e456267de476b5eb0eae
Date:   Sat Jan 2 11:03:12 2021 +0100

    Fix some warnings for Raspberry pi 4 (part2)

which changed that

Change-Id: I9a19d7d6bc1e4115ffffbe32d8d62be5d275d500
---
 bridges/source/cpp_uno/gcc3_linux_arm/uno2cpp.cxx | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bridges/source/cpp_uno/gcc3_linux_arm/uno2cpp.cxx b/bridges/source/cpp_uno/gcc3_linux_arm/uno2cpp.cxx
index 8533415ed087..7a8adad90ead 100644
--- a/bridges/source/cpp_uno/gcc3_linux_arm/uno2cpp.cxx
+++ b/bridges/source/cpp_uno/gcc3_linux_arm/uno2cpp.cxx
@@ -251,7 +251,7 @@ void callVirtualMethod(
     {
         // 8-bytes aligned
         sal_uInt32 nStackBytes = ( ( nStack + 1 ) >> 1 ) * 8;
-        sal_uInt32 *stack = static_cast<sal_uInt32 *>(__builtin_alloca( nStackBytes * sizeof(sal_uInt32)));
+        sal_uInt32 *stack = static_cast<sal_uInt32 *>(__builtin_alloca( nStackBytes ));
         memcpy( stack, pStack, nStackBytes );
     }
 
-- 
2.31.1

